batch10/assignment2/tamkeen-zarah.md

```diff
@@ -0,0 +1,43 @@
+# Q1) Share screenshot of the above Branch Protection of main branch
+
+![Branch Protection of main branch](https://user-images.githubusercontent.com/62700476/194774394-a514f13d-35b8-42e2-90a0-0504059f3fad.png)
+
+
+# Q2) Explain Docker Containers vs VMs
+
+## Docker Containers
+
+- OS level virtualisation
+- light weight
+- Fast
+- Simply isolated processes 
+- Memory efficient
+
+## VMs
+
+- Hardware level virtualisation
+- Shared apps
+- Dedicated memory
+
+# Q3) Explain Docker Architecture
+
+- Docker is a tool used for containerization
+- Sharing of images
+- It can create,ship and run containers
+- Runs as a single process
+
+# Q4) Write command to create an nginx container in detached mode with name assignment-2 running on host port 9090 and container port 80 on a custom network named assignment-2
+
```

Comment:
Also add command to create custom network and then run the container on that network